src/main/groovy/com/github/roroche/plantuml/urls/MainOutputUrlsFactory.groovy

```diff
@@ -0,0 +1,26 @@
+package com.github.roroche.plantuml.urls
+
+import org.gradle.api.Project
+
+import static com.github.roroche.plantuml.android.AndroidProjectType.*
+
+/***
+ * Factory for creating the correct output URLs based on the type of a project
+ */
+class MainOutputUrlsFactory {
+    static Urls.Wrap createMainOutputUrls(Project project) {
```

Comment:
So this logic can go to a `KotlinProject: SupportedProject` and it will be composed such as:

```kotlin
class KotlinProject(private val project: Project): SupportedProject {
  fun sourceUrls(): Urls {
    def debugTask = project.tasks.find {
      it.name.contains("compile") && it.name.contains("DebugKotlin")
    }
    return new MainOutputUrls(
      new FilesUrls(
        [debugTask.destinationDir].toSet()
      )
    )
  }
}

class AndroidApplication(project: Project): SupportedProject.Wrap(
  delegate = KotlinProject(project)
) {
  fun sourceUrls(): Urls {
    val ktSrcUrls = super.sourceUrls()
    // return the specific list + ktSrcUrls 
  }
}
```